test/suites/dev/test-xcm/test-xcm-erc20-transfer-two-ERC20.ts

```diff
@@ -0,0 +1,276 @@
+          await polkadotJs.query.system.account(ALITH_ADDRESS)
+        ).data.free.toBigInt();
+
+        // Destination as multilocation
+        const destination = [
+          // one parent
+          1,
+          // This represents X1(AccountKey20(BALTATHAR_ADDRESS, NetworkAny))
+          // AccountKey20 variant (03) + the 20 bytes account + Any network variant (00)
+          ["0x03" + BALTATHAR_ADDRESS.slice(2) + "00"],
+        ];
+
+        const currency1 = [erc20ContractAddress1, amountTransferred];
+        const currency2 = [erc20ContractAddress2, amountTransferred];
+
+        const rawTx = await context.writePrecompile!({
+          precompileName: "Xtokens",
+          functionName: "transferMultiCurrencies",
+          args: [
+            // address of the multiassets
+            [currency1, currency2],
+            // index fee
+            1n,
+            // Destination as multilocation
+            destination,
+            // weight
+            500_000_000n,
+          ],
+          rawTxOnly: true,
+        });
+
+        const { result } = await context.createBlock(rawTx);
+        expectEVMResult(result!.events, "Succeed");
+
+        const receipt = await w3.eth.getTransactionReceipt(result!.hash);
+        const gasPrice = receipt.effectiveGasPrice;
+        const fees = BigInt(receipt.gasUsed) * BigInt(gasPrice!);
+
+        const balanceAfter = (
+          await polkadotJs.query.system.account(ALITH_ADDRESS)
+        ).data.free.toBigInt();
+
+        // Fees should have been spent
+        expect(balanceAfter).to.equal(balanceBefore - fees);
+
+        // Erc20 tokens of the first contract should have been spent
+        expect(
+          await context.readContract!({
+            contractName: "ERC20WithInitialSupply",
+            contractAddress: erc20ContractAddress1 as `0x${string}`,
+            functionName: "balanceOf",
+            args: [ALITH_ADDRESS],
+          })
+        ).equals(ERC20_TOTAL_SUPPLY - amountTransferred);
+
+        // Erc20 tokens of the second contract should have been spent
+        expect(
+          await context.readContract!({
+            contractName: "ERC20WithInitialSupply",
+            contractAddress: erc20ContractAddress2 as `0x${string}`,
+            functionName: "balanceOf",
+            args: [ALITH_ADDRESS],
+          })
+        ).equals(ERC20_TOTAL_SUPPLY - amountTransferred);
+      },
+    });
+    it({
+      id: "T02",
+      title: "Should be able to transfer two ERC20 through incoming XCM message",
+      test: async function () {
+        const paraId = 888;
+        const paraSovereign = sovereignAccountOfSibling(context, paraId);
+        const amountTransferredOf1 = 1_000_000n;
+        const amountTransferredOf2 = 2_000_000n;
+
+        // Get pallet indices
+        const metadata = await polkadotJs.rpc.state.getMetadata();
+        const balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
```

Comment:
Same comment as before, interact with object directly without casting in order to preserve type safety